state-chain/pallets/cf-elections/src/electoral_systems/tests/delta_based_ingress.rs

```diff
@@ -0,0 +1,383 @@
+			asset: Asset::Sol,
+			total_ingressed: 500u64,
+			block_number: 900u64,
+			close_block: 1_000u64,
+		},
+		DepositChannel {
+			account: 2u32,
+			asset: Asset::SolUsdc,
+			total_ingressed: 1_000u64,
+			block_number: 900u64,
+			close_block: 2_000u64,
+		},
+	])
+}
+
+fn channel_state_final() -> DepositChannels {
+	DepositChannels::new(vec![
+		DepositChannel {
+			account: 1u32,
+			asset: Asset::Sol,
+			total_ingressed: 2_000u64,
+			block_number: 900u64,
+			close_block: 1_000u64,
+		},
+		DepositChannel {
+			account: 2u32,
+			asset: Asset::SolUsdc,
+			total_ingressed: 4_000u64,
+			block_number: 900u64,
+			close_block: 2_000u64,
+		},
+	])
+}
+
+fn channel_state_closed() -> DepositChannels {
+	DepositChannels::new(vec![
+		DepositChannel {
+			account: 1u32,
+			asset: Asset::Sol,
+			total_ingressed: 2_000u64,
+			block_number: 1_000u64,
+			close_block: 1_000u64,
+		},
+		DepositChannel {
+			account: 2u32,
+			asset: Asset::SolUsdc,
+			total_ingressed: 4_000u64,
+			block_number: 2_000u64,
+			close_block: 2_000u64,
+		},
+	])
+}
+
+fn with_setup(initial_elections: Option<DepositChannels>) -> TestSetup<SimpleDeltaBasedIngress> {
+	let initial_elections = initial_elections.unwrap_or(initial_channel_state());
+	TestSetup::<_>::default()
+		.with_initial_election_state(
+			1u32,
+			initial_elections.to_properties(),
+			initial_elections.to_state(),
+		)
+		.with_initial_state_map(initial_elections.to_state_map().into_iter().collect::<Vec<_>>())
+}
+
+register_checks! {
+	SimpleDeltaBasedIngress {
+		started_at_default_state(pre_finalize, _post) {
+			assert_eq!(
+				pre_finalize.unsynchronised_state_map,
+				to_raw_state_map(initial_channel_state().to_state_map()),
+				"Expected initial state incorrect pre-finalization."
+			);
+		},
+		ended_at_ingressed_state(_pre, post_finalize) {
+			assert_eq!(
+				post_finalize.unsynchronised_state_map,
+				to_raw_state_map(channel_state_ingressed().to_state_map()),
+				"Expected ingressed state incorrect after finalization."
+			);
+		},
+		deposit_channel_states_final(pre_finalize, post_finalize) {
+			assert_eq!(
+				pre_finalize.unsynchronised_state_map,
+				to_raw_state_map(channel_state_ingressed().to_state_map()),
+				"Expected ingressed state incorrect pre-finalization."
+			);
+			assert_eq!(
+				post_finalize.unsynchronised_state_map,
+				to_raw_state_map(channel_state_final().to_state_map()),
+				"Expected final state incorrect after finalization."
+			);
+		},
+	}
+}
+
+#[test]
+fn trigger_ingress_on_consensus() {
+	with_setup(None)
+		.build_with_initial_election()
+		.then(|| {
+			AMOUNT_INGRESSED.with(|ingresses| {
+				assert_eq!(ingresses.clone().into_inner().len(), 0);
+			});
+		})
+		.force_consensus_update(ConsensusStatus::Gained {
+			most_recent: None,
+			new: channel_state_ingressed().to_state(),
+		})
+		.test_on_finalize(
+			&900,
+			|_| (),
+			vec![Check::started_at_default_state(), Check::ended_at_ingressed_state()],
+		)
+		.then(|| {
```

Comment:
I recently had a PR merge to main which allows these to be turned into `Check`s by allowing passing a parameter to a check. Seems we dupe this ingress amount check, so we could definitely make a Check with a parameter for that. 
the PR that adds it, and converts some existing Checks: https://github.com/chainflip-io/chainflip-backend/pull/5441

Reply:
I ended up keeping this `then` function, because I wanted to test `open_channel`. 
I thought opening new channels as part of "finalize checks" feels weird, but more critically, open_channel modifies the states of the Electoral system, (e.g. starts a new election) and I need the "identifier" field to refresh after an election has started. It would be easier to have this function where code that mutates Electoral state can execute.